### DISCUSSION TITLE
How can we solve the k8s.io/code-generator versioning issue?

### DISCUSSION THREAD

**Discussion by gvmw** [rabbitmq/messaging-topology-operator#256] [General]
If we run `go get -d github.com/rabbitmq/messaging-topology-operator@v1.2.1` then `k8s.io/code-generator` gets updated from `v0.21.4` to `v0.22.2`. This in turn bumps other dependencies which breaks many other things ([this comment has all the details]( https://github.com/knative-sandbox/eventing-rabbitmq/pull/492#issuecomment-963198259)).

If I downgrade `k8s.io/code-generator@v0.21.4` explicitly, this happens:

```
go get -d k8s.io/code-generator@v0.21.4
go get: downgraded github.com/rabbitmq/messaging-topology-operator v1.2.1 => v0.11.0
go get: downgraded k8s.io/code-generator v0.22.2 => v0.21.4
```

Could this repo pin `k8s.io/code-generator` to `v0.21.4`?

This is the context in which I'm trying to make this work: https://github.com/knative-sandbox/eventing-rabbitmq/pull/492

**Comment by gvmw** [comment]
I have found something interesting in https://github.com/rabbitmq/messaging-topology-operator/pull/87.

I think I have a fix, will open a PR when I have it all working locally.

**Reply by gvmw** [reply]
The PR is now in - https://github.com/rabbitmq/messaging-topology-operator/pull/257 - maybe we should convert this to an issue - WDYT?

**Reply by Zerpet** [reply]
Sounds good to me, I'll convert this to an issue. I don't think we have a strong requirement on the specific version for code generator.

**Reply by gvmw** [reply]
As soon as https://github.com/knative-sandbox/eventing-rabbitmq/pull/492 goes green, we can merge https://github.com/rabbitmq/messaging-topology-operator/pull/257
